dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/imports/programrule/executor/enrollment/SetMandatoryFieldExecutorTest.java

```diff
@@ -145,24 +149,100 @@ void shouldReturnNoErrorWhenMandatoryFieldIsPresentForEnrollmentUsingIdSchemeCod
   }
 
   @Test
-  void shouldReturnAnErrorWhenMandatoryFieldIsNotPresentForEnrollment() {
+  void shouldReturnAnErrorWhenCreatingEnrollmentAndMandatoryFieldIsNotPresent() {
     when(preheat.getIdSchemes()).thenReturn(TrackerIdSchemeParams.builder().build());
     when(preheat.getTrackedEntityAttribute(ATTRIBUTE_UID.getValue())).thenReturn(attribute);
+    Enrollment enrollmentWithMandatoryAttributeNOTSet = getEnrollmentWithMandatoryAttributeNOTSet();
+    Enrollment enrollmentWithMandatoryAttributeSet = getEnrollmentWithMandatoryAttributeSet();
     bundle.setEnrollments(
-        List.of(
-            getEnrollmentWithMandatoryAttributeSet(), getEnrollmentWithMandatoryAttributeNOTSet()));
+        List.of(enrollmentWithMandatoryAttributeSet, enrollmentWithMandatoryAttributeNOTSet));
+    bundle.setStrategy(enrollmentWithMandatoryAttributeNOTSet, TrackerImportStrategy.CREATE);
+    bundle.setStrategy(enrollmentWithMandatoryAttributeSet, TrackerImportStrategy.CREATE);
 
     Optional<ProgramRuleIssue> error =
-        executor.executeRuleAction(bundle, getEnrollmentWithMandatoryAttributeSet());
+        executor.executeRuleAction(bundle, enrollmentWithMandatoryAttributeSet);
 
     assertFalse(error.isPresent());
```

Comment:
This tests separate cases right? They should be independent tests if thats true (same for the other tests).